core/module/services.types.ts

```diff
@@ -10,41 +15,43 @@ export type ServiceName =
 export type ServiceInterface<T extends ServiceName> = T extends "oauth"
   ? AuthService
   : T extends "medical-api"
-  ? MedicalTranscription
-  : // TODO
-  T extends "templates"
-  ? never
-  : T extends "subscriptions"
-  ? SubscriptionService
-  : never;
+    ? MedicalTranscription
+    : // TODO
+      T extends "templates"
+      ? TemplateService
+      : T extends "subscriptions"
+        ? SubscriptionService
+        : never;
 
 export interface MedicalTranscription {
   transcribeAudio(audioFile: File): Promise<string>;
   generateReport(
     transcription: string,
     template?: string,
-    language?: string
+    language?: string,
   ): Promise<string>;
   processAudioAndGenerateReport(
     audioFile: File,
     template?: string,
-    language?: string
+    language?: string,
   ): Promise<{
-    report: string; transcription: string, time: {
-      transcription: number,
-      report: number,
-    }
+    report: string;
+    transcription: string;
+    time: {
+      transcription: number;
+      report: number;
+    };
```

Comment:
I think in my branch now is called GenesisReport.

No need to change here, in case ther eis conflict, take `GeneisReport`

We call Genesis all the object of our busness logic, as a standard, they may come from genesis,s upabase, or. node in the future